File: kolko&krzyzykpython/board.py
class Board:
    def __init__(self):
        self.grid = [["" for _ in range(3)] for _ in range(3)]

    def reset(self):
        self.grid = [["" for _ in range(3)] for _ in range(3)]

    def check_win(self, symbol):
        for i in range(3):
            if all(self.grid[i][j] == symbol for j in range(3)) or \
               all(self.grid[j][i] == symbol for j in range(3)):
                return True
        if all(self.grid[i][i] == symbol for i in range(3)) or \
           all(self.grid[i][2 - i] == symbol for i in range(3)):
            return True
        return False

    def is_full(self):
        return all(cell != "" for row in self.grid for cell in row)

File: kolko&krzyzykcs/Board.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace kolko_krzyzykcs
{
    public class Board
    {
        public string[,] Grid { get; set; } = new string[3, 3];

        public void Reset()
        {
            for (int i = 0; i < 3; i++)
                for (int j = 0; j < 3; j++)
                    Grid[i, j] = "";
        }

        public bool CheckWin(string symbol)
        {
            for (int i = 0; i < 3; i++)
            {
                if (Grid[i, 0] == symbol && Grid[i, 1] == symbol && Grid[i, 2] == symbol) return true;
                if (Grid[0, i] == symbol && Grid[1, i] == symbol && Grid[2, i] == symbol) return true;
            }
            if (Grid[0, 0] == symbol && Grid[1, 1] == symbol && Grid[2, 2] == symbol) return true;
            if (Grid[0, 2] == symbol && Grid[1, 1] == symbol && Grid[2, 0] == symbol) return true;

            return false;
        }

        public bool IsFull()
        {
            foreach (var cell in Grid)
                if (string.IsNullOrEmpty(cell))
                    return false;
            return true;
        }
    }

}
